app/models/post.rb

```diff
@@ -0,0 +1,14 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
+
+  def user_posts_counter_updater
+    author.update(posts_counter: author.posts.count)
+    author.posts.count
+  end
```

Comment:
- [Optional] Dear @porag-m06, for such a method which is expected to run every time we create a new post it's a better practice to apply the `after_save` helper. This way we can keep our data consistent and we don't need to call the `user_posts_counter_updater` method to update the author post counter.  [see here for more](https://guides.rubyonrails.org/active_record_callbacks.html)